src/main/java/com/iexec/worker/metric/ComputeDurationsService.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2023-2023 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.worker.metric;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.apache.commons.math3.stat.descriptive.DescriptiveStatistics;
+
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Simple service to store durations of a compute stage (pre, app or post-compute)
+ * and aggregates them (min, max and average).
+ */
+public class ComputeDurationsService {
+    private static final String EXPORTED_STAT_PREFIX = "iexec_";
+
+    private final Map<String, Long> durationPerChainTaskId = new LinkedHashMap<>();
+    private final DescriptiveStatistics statistics;
+
+    public ComputeDurationsService(MeterRegistry registry,
+                                   String workerWalletAddress,
+                                   String context,
+                                   int statisticsWindow) {
+        this.statistics = new DescriptiveStatistics(statisticsWindow);
```

Comment:
Just for the sake of precision here (seems a little more compliant with the `setWindowSize` call performed from `DescriptiveStatistics` constructor).
What do you think of ?
```suggestion
                                   int statisticsWindowSize) {
        this.statistics = new DescriptiveStatistics(statisticsWindowSize);
```
or just
```suggestion
                                   int windowSize) {
        this.statistics = new DescriptiveStatistics(windowSize);
```

Reply:
Good idea. Please see [ecad540](https://github.com/iExecBlockchainComputing/iexec-worker/pull/559/commits/ecad540de39dd1319f016148ad2c9f2e3441860a).